src/Catel.MVVM/Catel.MVVM.Shared/Services/PleaseWaitService.cs

```diff
@@ -150,14 +152,15 @@ public void Hide()
         /// <remarks></remarks>
         public void Push(string status = "")
         {
-            UpdateStatus(status);
-
-            _showCounter++;
-
-            if (_showCounter > 0)
+            if (_showCounter <= 0)
             {
-                Show();
```

Comment:
This way the show counter will always be 1 because the Show method will reset to 1.